### ISSUE TITLE
Handle New England states

### ISSUE THREAD

**Issue by jeremyjbowers** [newsdev/elex#123]
For states in New England, `level=ru` will return townships. Thankfully, those townships have a FIPS code attached for the county they belong to. We should use these FIPS to create a county-level rollup that sums the votes counts and generates new pcts.

- [ ] Identify if this race needs to have the rollups applied.
- [ ] Implement a reporting unit-level method to loop over results and build new county objects based on FIPS code.
- [ ] Get county object's `reportingunitname` from `maps.py` since we'll only have a FIPS code.
- [ ] Test that this works for both New England states (2012-11-06 has ME Senate) and non-New England states.

**Comment by jeremyjbowers** [comment]
```
nosetests tests.test_new_england_reporting_units
E
======================================================================
ERROR: New England states should create county-level rollups.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/jbowers/src/nyt/elex/tests/__init__.py", line 13, in setUp
    self.race_objs = e.get_race_objects(self.raw_races)
  File "/Users/jbowers/src/nyt/elex/elex/parser/api.py", line 800, in get_race_objects
    return [Race(**r) for r in parsed_json['races']]
  File "/Users/jbowers/src/nyt/elex/elex/parser/api.py", line 595, in __init__
    self.set_reportingunits()
  File "/Users/jbowers/src/nyt/elex/elex/parser/api.py", line 66, in set_reportingunits
    obj = ReportingUnit(**reportingunit_dict)
  File "/Users/jbowers/src/nyt/elex/elex/parser/api.py", line 420, in __init__
    self.set_new_england_counties()
  File "/Users/jbowers/src/nyt/elex/elex/parser/api.py", line 461, in set_new_england_counties
    results[c.unique_id][c.reportingunitname] = maps.FIPS_TO_STATE[c.statepostal][c.fipscode]
KeyError: None

----------------------------------------------------------------------
Ran 1 test in 0.047s

FAILED (errors=1)
```

**Comment by jeremyjbowers** [comment]
On the plus side, I've verified this approach with the AP's Tracy Lewis, so that's good.

**Comment by jeremyjbowers** [comment]
Getting closer.
```
nosetests tests.test_new_england_reporting_units
F.FF..
======================================================================
FAIL: test_florida_counties (tests.test_new_england_reporting_units.TestCandidate)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/jbowers/src/nyt/elex/tests/test_new_england_reporting_units.py", line 42, in test_florida_counties
    self.assertEqual(len(florida_counties), len(florida_results))
AssertionError: 67 != 68

======================================================================
FAIL: test_maine_counties (tests.test_new_england_reporting_units.TestCandidate)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/jbowers/src/nyt/elex/tests/test_new_england_reporting_units.py", line 56, in test_maine_counties
    self.assertEqual(len(maine_counties), count_maine_counties)
AssertionError: 0 != 16

======================================================================
FAIL: test_maine_townships (tests.test_new_england_reporting_units.TestCandidate)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/jbowers/src/nyt/elex/tests/test_new_england_reporting_units.py", line 49, in test_maine_townships
    self.assertEqual(len(maine_townships), 500 - count_maine_counties)
AssertionError: 499 != 484

----------------------------------------------------------------------
Ran 6 tests in 10.358s

FAILED (failures=3)
```

**Comment by jeremyjbowers** [comment]
Now down to two fails. And they're failing correctly.

```
nosetests tests.test_new_england_reporting_units
..FF..
======================================================================
FAIL: test_maine_counties (tests.test_new_england_reporting_units.TestCandidate)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/jbowers/src/nyt/elex/tests/test_new_england_reporting_units.py", line 58, in test_maine_counties
    self.assertEqual(len(maine_counties), count_maine_counties)
AssertionError: 0 != 16

======================================================================
FAIL: test_maine_townships (tests.test_new_england_reporting_units.TestCandidate)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/jbowers/src/nyt/elex/tests/test_new_england_reporting_units.py", line 51, in test_maine_townships
    self.assertEqual(len(maine_townships), count_maine_results_minus_state - count_maine_counties)
AssertionError: 499 != 483

----------------------------------------------------------------------
Ran 6 tests in 10.673s
```

**Comment by jeremyjbowers** [comment]
Okay, this is happening because I'm adding new CandidateReportingUnits under the existing ReportingUnits. But I really need to be creating new ReportingUnits and add the new CandidateReportingUnits under those.

TL;DR I need to move the function that does this to the Race object.
